app/src/backend/environment/hooks/useEnvironmentManager.ts

```diff
@@ -18,8 +18,9 @@ import { toast } from "utils/Toast";
 
 let unsubscribeListener: () => void = null;
 
-const useEnvironmentManager = () => {
+const useEnvironmentManager = (initListenerAndFetcher: boolean = false) => {
```

Comment:
Instead of `initListenerAndFetcher` what if we check if listener has been attached or not and attach it only once?